language/diem-tools/transaction-replay/src/lib.rs

```diff
@@ -206,6 +208,32 @@ impl DiemDebugger {
         Ok(modules)
     }
 
+    pub fn annotate_events(&self, event_key: &EventKey, start_seq: u64, limit: u64) -> Result<()> {
```

Comment:
This code looks potentially useful--thoughts about including the logic for returning a `Vec<AnnotatedMoveStruct>` inside the client code directly instead of in transaction-replay?

CC @bmwill for feedback on this as well.

Reply:
The reason I implemented this way is because there isn't such thing as `AnnotatedEvent` and simply use `AnnotatedMoveStruct` will lose information like `event_index`, proof, etc. I can expose a separate api for annotating each individual event tho.

Reply:
If you take a look at the client code, it has an `Event<T>` [type](https://github.com/diem/diem/blob/main/sdk/client/src/move_deserialize.rs#L17) that you can use for wrapping the payload in `T` while retaining the index etc. I think you will need to implement the `MoveResource` trait for `AnnotatedMoveStruct` in order to do this, but that shouldn't be a problem.

Reply:
I don't think the Event<T> would work as MoveResource is really a rust struct and would require users to write Rust scaffolding code to decode event payload, which isn't sth that I intended for the AnnotatedMoveStruct. On the other hand, I implemented the api that you mentioned previously to batch annotate payloads of events into a AnnotatedMoveStruct. Will that work?

Reply:
Ah, my question was poorly posed; sorry about that. What I meant to be asking about is a variant of [this](https://github.com/diem/diem/blob/main/sdk/client/src/client.rs#L272) that does not have a generic type parameter + returns an `Event<AnnotatedMoveStruct>`